prototype_source/aot_inductor_tutorial.py

```diff
@@ -0,0 +1,141 @@
+#
+# .. warning::
+#
+#     AOTInductor and its related features are in prototype status and are
+#     subject to backwards compatibility breaking changes. It is not meant to be
+#     used in any production environment.
+#
+#
+# AOTInductor is a specialized version of
+# `TorchInductor <https://dev-discuss.pytorch.org/t/torchinductor-a-pytorch-native-compiler-with-define-by-run-ir-and-symbolic-shapes/747>`__
+# which takes an exported PyTorch model, optimizes it, and generates a shared
+# library. These compiled artifacts can be deployed to non-Python environments,
+# which are commonly used for inference deployments.
+#
+# AOTInductor is a vital component along the
+# `export <https://pytorch.org/docs/main/export.html>`__ path as it provides a way
+# to run an exported model without a Python runtime.
+#
+# In this tutorial, you will learn how take a PyTorch model, export it to a
+# compilation graph, pass it to AOTInductor to be compiled into a shared binary,
+# and run this binary.
+#
+# ..note:: a modern NVIDIA GPU (H100, A100, or V100) is recommended for this
+# tutorial in order to reproduce the speedup numbers shown below and documented
+# elsewhere.
+
+######################################################################
+# Entrypoint
+# ----------
+#
+# The entrypoint to AOTInductor is through the ``torch._export.aot_compile``
+# function. Note that this function is still in prototype stage and is subject to change.
+#
+# The signature is:
+#
+# .. code:: python
+#
+#   torch._export.aot_compile(
+#       module: Callable,
+#       args: Tuple[Any, ...],
+#       kwargs: Optional[Dict[str, Any]] = None,
+#       *,
+#       dynamic_shapes: Optional[Dict[str, Any]] = None,
+#       options: Optional[Dict[str, Any]] = None,
+#   ) -> Tuple[str, bytes]
+#
+# This function does the following:
+#
+#    1. Exports the given model into a compilation graph through the
+#      `torch.export <https://pytorch.org/docs/main/export.html>`__ workflow. If
+#      there are any errors when exporting the model, please refer to the
+#      `torch.export <https://pytorch.org/docs/main/export.html>`__ documentation.
+#    2. Compiles the exported program using TorchInductor into a shared library.
+#    3. Returns the path to the shared library.
+#
+
+import torch
+import torch._export
+import torch.utils._pytree as pytree
+from typing import Any, Tuple, Mapping
+
+class M(torch.nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        self.branch1 = torch.nn.Sequential(
+            torch.nn.Linear(64, 32), torch.nn.ReLU()
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return self.branch1(x)
+
+def _normalize_inputs(example_inputs) -> Tuple[Tuple[Any], Mapping[str, Any]]:
+    if isinstance(example_inputs, dict):
+        return (), example_inputs
+    else:
+        return example_inputs, {}
+
+with torch.no_grad():
+    m = M().to("cuda")
```

Comment:
Will be good to make the device configurable / passed as a variable to use the example for cpu / mps and other device types

Reply:
I don't think we support the other device types well to include in this tutorial (cc @desertfire)

Reply:
We can run this simple one on cpu, but AOTInductor for cpu is largely untested, and it won't be our focus before PTC.